_posts/2022-11-17-Multiple-Authentication.md

```diff
@@ -0,0 +1,51 @@
+4. Support identity redundancy across multiple IDPs to ensure high availability
+5. Simplify troubleshooting external IDP related issues
+
+## Use cases
+
+Customers using dedicated third-party enterprise-based IDPs to manage OpenSearch Dashboards user identities can now use OpenSearch as an Internal IDP for admin, security, and privileged accounts, and a single enterprise-based external IDP to maintain authentication information for general Dashboards user accounts. For larger enterprises that may have redundancy needs, they can enable multiple enterprise-based IDPs, for example, an enterprise OIDC IDP, and an enterprise SAML IDP.
+
+An additional use case for this that hasn’t been possible before is to add multiple external IDPs to mix and match needs. For example, OpenSearch can serve as an Internal IDP to maintain authentication information for admin, and security accounts, an Enterprise based IDP for privileged and analyst accounts, and a Social-based IDP for limited read-only accounts that can only view specific data.
+
+We would love to hear any additional use-cases you have that this feature can meet!
+
+## How can you use this feature?
+
+OpenSearch Dashboards admins can enable single or multiple authentication options on demand and customize the integrated Login UI by editing opensearch_dashboards.yml. For the essential settings required to setup multiple authentication, see [**Multiple Authentication Options**]({{site.url}}{{site.baseurl}}/security-plugin/configuration/multi-auth/). Once these options are configured, the Login UI will be automatically updated. For example, the login UI below shows both basic authentication and SSO enabled:
+
+![Default Login UI](/assets/media/blog-images/2022-11-14-multiple-authentication/default-login-ui.png){: .img-fluid}
+
+## How can you use this feature?
+
+When Dashboards bootstraps and loads security settings from `opensearch-dashboards.yml`, the Dashboards client can tell whether Single-Authentication Mode or Multiple-Authentication Mode is enabled by evaluating the Multiple Authentication feature flag and authentication type setting.
+
+* If **Single-Authentication Mode** is detected, only authentication handler for corresponding authentication type defined by *`attribute:opensearch_security.auth.type`* will be registered with the client. All the login requests flow to the dedicated IDP to complete identification and authentication.
+* If **Multiple-Authentication Mode** is detected, OSD will evaluate authentication handlers required for each individual authentication type defined by *`attribute:opensearch_security.auth.type`* and form a compound authentication handler, which will be registered with OSD client instead. This compound authentication handler is named as multi-auth handler. Multi-auth handler is able to redirect the authentication request to various IDP endpoints for authentication interchangeably based on the authentication type defined within Login request. 
```

Comment:
```suggestion
* If **Multiple-Authentication Mode** is detected, OSD will evaluate authentication handlers required for each individual authentication type defined by *`attribute:opensearch_security.auth.type`* and form a compound authentication handler, which will be registered with OSD client instead. This compound authentication handler is named multi-auth handler. Multi-auth handler is able to redirect the authentication request to various IdP endpoints for authentication interchangeably based on the authentication type defined within the login request. 
```

Reply:
Addressed as per suggestion. Thanks!

Reply:
1) I believe the first mention of "OSD" is different from "OSD client" that comes later in the sentence. 

2) In the text before the bullet points, it refers to an "authentication type setting." So I think we should properly name these "... the *`attribute:opensearch_security.auth.type`* setting ..." in both bullet points.

3) The sentences are long in both of these bullet points, and they're in the passive voice. I hope I'm not confusing things. But we might reword these (with help from Priscilla to get the proper names correct, i.e., OSD) like so:

- If **Single-Authentication Mode** is detected, the system [OSD?] registers a single authentication handler with the client based on the authentication type defined by the *`attribute:opensearch_security.auth.type`* setting. After that, all login requests flow to the dedicated IdP to complete identification and authentication.

- If **Multiple-Authentication Mode** is detected, the system [OSD?] evaluates all authentication handlers required for the multiple authentication types defined by the *`attribute:opensearch_security.auth.type`* setting and forms a compound authentication handler. The system then registers this compound authentication handler with the OSD client. The compound authentication handler is able to redirect the authentication request to various IdP endpoints for authentication interchangeably based on the authentication type defined within the login request.

Reply:
Thank for the comments. Addressed as per suggestion.